source/common/router/delegating_route_impl.h

```diff
@@ -25,9 +28,13 @@ class DelegatingRoute : public Router::Route {
   const Router::Decorator* decorator() const override;
   const Router::RouteTracing* tracingConfig() const override;
   const Router::RouteSpecificFilterConfig* perFilterConfig(const std::string&) const override;
+  const envoy::config::core::v3::Metadata& metadata() const override { return metadata_; }
+  const Envoy::Config::TypedMetadata& typedMetadata() const override { return typed_metadata_; }
 
 private:
   const Router::RouteConstSharedPtr base_route_;
+  const envoy::config::core::v3::Metadata metadata_;
+  const Envoy::Config::TypedMetadataImpl<Envoy::Config::TypedMetadataFactory> typed_metadata_;
```

Comment:
This should be calling the parent methods, not returning empty.

Reply:
I accidentally ignored it.